aws/table_aws_pricing_product.go

```diff
@@ -0,0 +1,274 @@
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxItems := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 20 {
+				maxItems = 20
+			} else {
+				maxItems = limit
+			}
+		}
+	}
+
+	equalQual := d.KeyColumnQuals
+	input := pricing.GetProductsInput{
+		ServiceCode:   aws.String(equalQual["service_code"].GetStringValue()),
+		FormatVersion: aws.String("aws_v1"),
+		MaxResults:    maxItems,
+	}
+
+	type OfferOutput struct {
+		_              struct{} `type:"structure"`
+		PriceDimension *PriceDimension
+		Term           string
+		EffectiveDate  *time.Time
+		OfferTermCode  *string
+		TermAttributes *TermAttributes
+	}
+
+	type PriceOutput struct {
+		_               struct{} `type:"structure"`
+		Product         *Product
+		ServiceCode     *string
+		Offer           *OfferOutput
+		Version         *string
+		PublicationDate *time.Time
+	}
+
+	filters := buildPricingFilter(d.Quals)
+	if len(filters) > 0 {
+		input.Filters = filters
+	}
+
+	paginator := pricing.NewGetProductsPaginator(svc, &input, func(o *pricing.GetProductsPaginatorOptions) {
+		o.Limit = maxItems
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_pricing_product.listPricingProduct", "api_error", err)
+			return nil, err
+		}
+
+		for _, priceList := range output.PriceList {
+			var priceListObject PriceList
+
+			err = json.Unmarshal([]byte(priceList), &priceListObject)
+			if err != nil {
+				plugin.Logger(ctx).Error("listPricingProduct unmarshal", "err", err)
+				return nil, err
+			}
+
+			for term, termDetail := range priceListObject.Terms {
+				for _, offer := range *termDetail {
+					for _, priceDimension := range offer.PriceDimensions {
+						d.StreamListItem(ctx, PriceOutput{
+							Product:         priceListObject.Product,
+							ServiceCode:     priceListObject.ServiceCode,
+							Version:         priceListObject.Version,
+							PublicationDate: priceListObject.PublicationDate,
+							Offer:           &OfferOutput{PriceDimension: priceDimension, Term: term, EffectiveDate: offer.EffectiveDate, OfferTermCode: offer.OfferTermCode, TermAttributes: offer.TermAttributes},
+						})
+					}
+				}
+			}
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+func extractPrice(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	priceMap := d.Value.(map[string]*string)
+
+	for _, units := range priceMap {
+		price, err := strconv.ParseFloat(*units, 32)
+		return price, err
+	}
+	return nil, nil
+}
+
+func extractCurrency(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	priceMap := d.Value.(map[string]*string)
+
+	for currency := range priceMap {
```

Comment:
Maybe I'm not understanding this code well, but why do we have a for loop and then return the first item we get?